src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PhysicalProjectTreeStorage.cs

```diff
@@ -12,27 +12,35 @@ internal class PhysicalProjectTreeStorage : IPhysicalProjectTreeStorage
     {
         private readonly Lazy<IFileSystem> _fileSystem;
         private readonly Lazy<IFolderManager> _folderManager;
-        private readonly Lazy<IPhysicalProjectTree> _projectTree;
+        private readonly Lazy<IProjectTreeService> _treeService;
+        private readonly Lazy<IProjectTreeProvider> _treeProvider;
         private readonly UnconfiguredProject _unconfiguredProject;
 
         [ImportingConstructor]
-        public PhysicalProjectTreeStorage(Lazy<IPhysicalProjectTree> projectTree, Lazy<IFileSystem> fileSystem, Lazy<IFolderManager> folderManager, UnconfiguredProject unconfiguredProject)
+        public PhysicalProjectTreeStorage([Import(ExportContractNames.ProjectTreeProviders.PhysicalProjectTreeService)]Lazy<IProjectTreeService> treeService,
+                                          [Import(ExportContractNames.ProjectTreeProviders.PhysicalViewTree)]Lazy<IProjectTreeProvider> treeProvider,
+                                          Lazy<IFileSystem> fileSystem, 
+                                          Lazy<IFolderManager> folderManager, 
+                                          UnconfiguredProject unconfiguredProject)
         {
             Requires.NotNull(fileSystem, nameof(fileSystem));
             Requires.NotNull(folderManager, nameof(folderManager));
-            Requires.NotNull(projectTree, nameof(projectTree));
+            Requires.NotNull(treeService, nameof(treeService));
```

Comment:
MEF Components are always expected to be non-null(Unless we specify AllowDefault attribute to true). These `null` checks and related tests are not necessary